File: Controllers/serviceController.js
const { ServiceItem } = require('../models/serviceItem');

const services = [
  new ServiceItem(1, 'Food Drive', 'Pack and distribute food to families in need.'),
  new ServiceItem(2, 'Tree Planting', 'Plant trees in the local community park.')
];

function sendJson(res, statusCode, payload) {
  res.writeHead(statusCode, { 'Content-Type': 'application/json' });
  res.end(JSON.stringify(payload));
}

function getAllServices(req, res) {
  sendJson(res, 200, services);
}

function getServiceById(req, res, id) {
  const item = services.find((service) => service.id.toString() === id);
  if (!item) {
    sendJson(res, 404, { message: 'Service item not found' });
    return;
  }

  sendJson(res, 200, item);
}

function createService(req, res) {
  let body = '';
  req.on('data', (chunk) => {
    body += chunk;
  });

  req.on('end', () => {
    try {
      const data = body ? JSON.parse(body) : {};
      const newItem = new ServiceItem(
        Date.now(),
        data.name || 'New Service',
        data.description || 'Added from API'
      );
      services.push(newItem);
      sendJson(res, 201, newItem);
    } catch (error) {
      sendJson(res, 400, { message: 'Invalid JSON payload' });
    }
  });
}

module.exports = { getAllServices, getServiceById, createService };

File: Controllers/ServiceController.cs
using System.Collections.Generic;
using System.Web;
using System.Web.Script.Serialization;
using ServiceDayBackend.Models;

namespace ServiceDayBackend.Controllers
{
    public class ServiceController
    {
        private readonly List<ServiceItem> _items = new List<ServiceItem>
        {
            new ServiceItem { Id = 1, Name = "Food Drive", Description = "Pack and distribute food to families in need." },
            new ServiceItem { Id = 2, Name = "Tree Planting", Description = "Plant trees in the local community park." }
        };

        public void HandleRequest(HttpContext context, string id)
        {
            context.Response.ContentType = "application/json";
            var serializer = new JavaScriptSerializer();

            if (context.Request.HttpMethod == "GET")
            {
                if (string.IsNullOrWhiteSpace(id))
                {
                    context.Response.Write(serializer.Serialize(_items));
                    return;
                }

                var item = _items.Find(x => x.Id.ToString() == id);
                if (item != null)
                {
                    context.Response.Write(serializer.Serialize(item));
                    return;
                }

                context.Response.StatusCode = 404;
                context.Response.Write(serializer.Serialize(new { message = "Service item not found" }));
                return;
            }

            if (context.Request.HttpMethod == "POST")
            {
                context.Response.StatusCode = 201;
                context.Response.Write(serializer.Serialize(new { message = "Service created successfully" }));
                return;
            }

            context.Response.StatusCode = 405;
            context.Response.Write(serializer.Serialize(new { message = "Method not allowed" }));
        }
    }
}
